akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -228,14 +238,19 @@ class ClusterClient(settings: ClusterClientSettings) extends Actor with Stash wi
     case ActorIdentity(_, Some(receptionist)) ⇒
       log.info("Connected to [{}]", receptionist.path)
       scheduleRefreshContactsTick(refreshContactsInterval)
-      unstashAll()
+      sendBuffered(receptionist)
```

Comment:
Yes, full flow control would be nice, but that is not in scope now, in my opinion. The buffer is intended to be small (we could set a max cap?).